flink-connector/src/main/java/org/apache/flink/connector/delta/sink/committer/DeltaGlobalCommitter.java

```diff
@@ -241,6 +312,19 @@ private String resolveAppId(List<DeltaGlobalCommittable> globalCommittables) {
         return actions;
     }
 
+    /**
+     * Prepares {@link Metadata} object for current transaction
+     *
+     * @return {@link Metadata} object for current transaction
+     */
+    private Metadata prepareMetadata() {
+        StructType dataStreamSchema = new SchemaConverter().toDeltaFormat(rowType);
+        return new Metadata
+            .Builder()
+            .schema(dataStreamSchema)
+            .build();
```

Comment:
i think this might be wrong. you have to create a metadata object that has all the other non-schema fields populated correctly based on the current metadata fields.

Reply:
@scottand-db we discussed that this can be a clear improvement to the existing APIs ... the caller should have to be so responsible for populating things unrelated to its intentions.

Reply:
until that happens. in the meantime, can atleast write the code to basically clone the existing snapshot.metadata into the new metadata object? we will simplify this later.

Reply:
You mean like using up all of the builder's methods ? (except of `createdTime` which is setting up current system time) Or maybe using the `Metadata`'s constructor directly?
I'm putting it this way so please do a double check on this discussion.
```
    /**
     * Prepares {@link Metadata} object for current transaction
     *
     * @param transactionMetadata {@link Metadata} object referring to the metadata of the table's
     *                            latest version as of this transaction's instantiation
     * @return {@link Metadata} object for current transaction
     */
    private Metadata prepareMetadata(Metadata transactionMetadata) {
        StructType dataStreamSchema = new SchemaConverter().toDeltaFormat(rowType);
        return new Metadata
            .Builder()
            .id(transactionMetadata.getId())
            .name(transactionMetadata.getName())
            .description(transactionMetadata.getDescription())
            .format(transactionMetadata.getFormat())
            // below line will be changed in the next PR 
            .partitionColumns(transactionMetadata.getPartitionColumns())  
            .configuration(transactionMetadata.getConfiguration())
            .schema(dataStreamSchema)
            .build();
    }
```

I agree that simple constructor for the builder accepting previous `Metadata` object would be helpful.